haystack/components/generators/chat/fallback.py

```diff
@@ -202,11 +217,12 @@ async def run_async(
         """
         failed: list[str] = []
         last_error: Union[BaseException, None] = None
+        callback = streaming_callback or self.streaming_callback
```

Comment:
Should we be doing something like we do in our other chat generators?
```python
from haystack.dataclasses import select_streaming_callback

        callback = select_streaming_callback(
            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=True
        )
```

Reply:
ok, I didn't know this pattern, I'll check

Reply:
Test, how? 🤔